src/zope/i18nmessageid/message.py

```diff
@@ -31,18 +31,37 @@ class Message(unicode):
     message id itself implicitly serves as the default text.
     """
 
-    __slots__ = ('domain', 'default', 'mapping', '_readonly')
+    __slots__ = (
+        'domain', 'default', 'mapping', '_readonly',
+        'msgid_plural', 'default_plural', 'number'
+    )
 
-    def __new__(cls, ustr, domain=None, default=None, mapping=None):
+    def __new__(cls, ustr, domain=None, default=None, mapping=None,
+                msgid_plural=None, default_plural=None, number=None):
         self = unicode.__new__(cls, ustr)
         if isinstance(ustr, self.__class__):
             domain = ustr.domain and ustr.domain[:] or domain
             default = ustr.default and ustr.default[:] or default
             mapping = ustr.mapping and ustr.mapping.copy() or mapping
+            msgid_plural = (
+                ustr.msgid_plural and ustr.msgid_plural[:] or msgid_plural)
+            default_plural = (
+                ustr.default_plural and ustr.default_plural[:]
+                or default_plural)
+            number = ustr.number is not None and ustr.number or number
             ustr = unicode(ustr)
+
         self.domain = domain
         self.default = default
         self.mapping = mapping
+        self.msgid_plural = msgid_plural
+        self.default_plural = default_plural or msgid_plural
```

Comment:
Why the `or msgid_plural` logic here?  It's not done for `self.default`, and it's not done in https://github.com/zopefoundation/zope.i18n/pull/34#pullrequestreview-165702854, where you can do `translate("msgid", msgid_plural="msgids")` without providing defaults and it'll pass a default_plural of None to the underlying catalog translation machinery.